hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -464,42 +427,40 @@ public AutoCloseableLock acquireDatasetReadLock() {
    * Activate a volume to serve requests.
    * @throws IOException if the storage UUID already exists.
    */
-  private void activateVolume(
+  private synchronized void activateVolume(
```

Comment:
Just suggest to add annotation to describe that we use #synchronized to protect volume instance and use datasetLock to protect blockpool instance.